modules/core/arrow-core/src/main/kotlin/arrow/core/Either.kt

```diff
@@ -276,6 +276,23 @@ inline fun <A, B> EitherOf<A, B>.getOrHandle(crossinline default: (A) -> B): B =
 inline fun <A, B> EitherOf<A, B>.filterOrElse(crossinline predicate: (B) -> Boolean, crossinline default: () -> A): Either<A, B> =
   fix().fold({ Left(it) }, { if (predicate(it)) Right(it) else Left(default()) })
 
+/**
+ * * Returns [Either.Right] with the existing value of [Either.Right] if this is an [Either.Right] with a non-null value.
+ * The returned Either.Right type is not nullable.
+ * * Returns `Left(default())` if this is an [Either.Right] and the existing value is null
+ * * Returns [Either.Left] with the existing value of [Either.Left] if this is an [Either.Left].
+ *
+ * Example:
+ * ```
+ * Right(12).leftIfNull({ -1 }) // Result: Right(12)
+ * Right(null).leftIfNull({ -1 })  // Result: Left(-1)
+ *
+ * Left(12).leftIfNull({ -1 })      // Result: Left(12)
+ * ```
+ */
+inline fun <A, B> EitherOf<A, B?>.leftIfNull(crossinline default: () -> A): Either<A, B> =
+        fix().flatMap { if ( it == null) Left(default()) else Right(it) }
```

Comment:
I'm surprised our linter doesn't complain about the extra whitespace here

Reply:
U mean at the beginning. Yes, IntelliJ has added some extra here.

Reply:
Fixed. Before ```it == null``` as well.